octoprint_PrintTimeGenius/static/js/PrintTimeGenius.js

```diff
@@ -70,19 +70,25 @@ $(function() {
         }
       }
       return results;
-    }
+    };
 
-    self.FileList = ko.pureComputed(function() {
-      return self.recurseFiles(self.filesViewModel.allItems()).slice()
-          .sort(function(a,b) {
-            if (_.has(a, "gcodeAnalysis.progress") !=
```

Comment:
This sort was lost in the patch but I think that it is useful to have.  What do you think?

Reply:
Ah yes, I'll add that back. I'm not sure if that bit was present when I first came up with this patch or not

Reply:
looking back on this, I have found the sorting with the old version very confusing, as sorting by analysis progress mostly results in an unpredictable ordering, since the majority of files are always already fully analyzed. The code here keeps the same ordering as the file list in the main UI, which I think is much more natural and expected

Reply:
The fall back on line 82 below ensures that all the analyzed files are alphabetical.  So in the end you get all the unanalyzed files at the top, sorted by name, and then all the analyzed files below, sorted by name.  Users liked having the files that need analysis all grouped together so that they could be selected with a click-then-shift-click.  And the unanalyzed ones are at the top, because those are the ones that you'll most likely want to analyze.

Reply:
Good point! I'll put this back then :)

Reply:
I've restored the previous sorting